tests/installation/validation/ibft.pm

```diff
@@ -113,6 +112,9 @@ sub run {
     # Requires NICTYPE=user and backend/qemu.pm code to run
     $ibft_expected->{ethernet}->{mac} = get_required_var('NICMAC');
 
+    my $fqdn = testapi::get_required_var('WORKER_HOSTNAME');
+    $ibft_expected->{target}->{ip_addr} = script_output("dig +short $fqdn");
```

Comment:
this might work but WORKER_HOSTNAME allows to still use an IP address for which dig will not return any result. Also, why go the extra mile to pass a command to the SUT and then back? Here you don't actually want to test the behaviour of "dig", do you?

Reply:
@okurz 
I agree we need check WORKER_HOSTNAME is fqdn or ip address before do fqdn to ip action, but if you think use dig command on SUT is not good idea, then what's your suggestion to get map relation between WORKER_HOSTNAME & ip?

Reply:
> Can grains['fqdn'][0] [here](https://gitlab.suse.de/openqa/salt-states-openqa/-/merge_requests/766/diffs#61f80b79fa50900862d72f5ea55b805b8ca14fb8_154_153) contain an IP address?

No. But os-autoinst-distri-opensuse shouldn't code against how the infrastructure is managed

> but if you think use dig command on SUT is not good idea, then what's your suggestion to get map relation between WORKER_HOSTNAME & ip?

you can still use `dig` but why call it within the test VM and not just on the worker?

How about

```suggestion
use Socket;
…
    $ibft_expected->{target}->{ip_addr} = inet_ntoa(inet_aton($fqdn));
```

I tried it out with:

```
> print(inet_ntoa(inet_aton('openqa.suse.de')))
10.160.0.207
> print(inet_ntoa(inet_aton('10.160.0.207')))
10.160.0.207
```

Reply:
Thank you for the suggestion Oliver! I understand your reasoning, this works great